syntax = "proto3";
package CompE;

// ******************************************************************************************** //
message CMessage
{
    message Cstruct
    {
        int32 C_01 = 1;
        float C_02 = 2;
        double C_03 = 3;
    }
    repeated Cstruct C = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** //
message IMessage
{
    message Istruct
    {
        message Istructstruct
        {
        int32 I_01_01 = 1;
        float I_01_02 = 2;
        }
        repeated Istructstruct I = 1;
    }
    repeated Istruct Ist = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** //
message MMessage
{
    int64 M = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** // 
message CompEInMessage
{
    CMessage C = 1;
    IMessage I = 2;
    MMessage M = 3;
}
// ******************************************************************************************** //
